addons/website/tests/test_performance.py

```diff
@@ -4,7 +4,7 @@
 from odoo.tools import mute_logger
 from odoo.tests.common import HttpCase, tagged
 
-EXTRA_REQUEST = 2 - 1
+EXTRA_REQUEST = 2 * 3 - 1
```

Comment:
Didn't have time to properly investigate the "no cursor in matching" part of website to complete this PR but this is wrong.
The goal of those performance test and this `EXTRA_REQUEST` is to have the request number (second param of `_get_url_hot_query`) exactly the same as the one in real use case (outside test).
For instance, if the test is:
```py
    def test_30_perf_sql_queries_page_no_layout(self):
        # website.page with no call to layout templates
        self.page.arch = '<div>I am a blank page</div>'
        self.assertEqual(self._get_url_hot_query(self.page.url), 5)
        self.assertEqual(self._get_url_hot_query(self.page.url, cache=False), 5)
```
Then you are expecting when accessing manually to your browser and checking logs to see
`"GET /url HTTP/1.0" 200 - 5 0.003 0.012`
The `5` in the test reflects the reality.
But in tests, it's not 5 due to some specific testing behavior, as described in the comment.
The goal is obviously to have something human readable when debugging those tests or introducing a behavior which mess up with the perfs: you don't have to consider the testing tweaks, it's abstracted in this `EXTRA_REQUEST` variable.
Otherwise, it would be impossible to work with.

But here, actually, your diff make it looks like it's only changing the testing extra queries while actually it's skyrocketing the real query count:

| url | before | after |
| - | - | - |
|regular page | 7 | 11 |
|homepage | 7 | 13 |
|no layout page | 6 | 10 |

Note: I've checked in 15.0 and reality/`_get_url_hot_query` are correctly sync, but in 16.0 it seems like one or two test are off by (just) one query, I'll have a look to align those and see what's going on.

Reply:
Waiting for https://github.com/odoo/odoo/pull/119268 to be merged and ported to master before checking this test perf change.
Those tests are testing the most important thing we have for our website: the perfs. It was a bit too fragile, the PR is improving the checks.
I had to introduce 2 changes in the framework js sql logger tho, waiting for guidance/review there, didn't get any reply so far.